cristin-import/src/main/java/no/unit/nva/cristin/lambda/DoiDuplicateChecker.java

```diff
@@ -0,0 +1,63 @@
+package no.unit.nva.cristin.lambda;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.net.URI;
+import java.util.Optional;
+import no.unit.nva.commons.json.JsonUtils;
+import no.unit.nva.cristin.mapper.SearchResource2Response;
+import no.unit.nva.cristin.mapper.nva.exceptions.DuplicateDoiException;
+import no.unit.nva.publication.external.services.UriRetriever;
+import nva.commons.core.paths.UriWrapper;
+
+public class DoiDuplicateChecker {
+
+    private static final String SEARCH = "search";
+    private static final String RESOURCES2 = "resources2";
+    private static final String DOI = "doi";
+    private static final String APPLICATION_JSON = "application/json";
+    private final UriRetriever uriRetriever;
+    private final String apiHost;
+
+    public DoiDuplicateChecker(UriRetriever uriRetriever, String apiHost) {
+        this.uriRetriever = uriRetriever;
+        this.apiHost = apiHost;
+    }
+
+    public PublicationRepresentations throwIfDoiExists(PublicationRepresentations pubRep) {
+        return Optional.ofNullable(pubRep.getPublication().getEntityDescription().getReference().getDoi())
+                   .map(doi -> fetchNvaPublicationsByDoi(doi, pubRep))
+                   .orElse(pubRep);
+    }
+
+    private PublicationRepresentations fetchNvaPublicationsByDoi(URI doi, PublicationRepresentations pubRep) {
+        var request = constructSearchUri(doi);
+        return getResponseBody(request)
+                   .map(response -> checkForHits(response, doi, pubRep))
+                   .orElse(pubRep);
+    }
+
+    private PublicationRepresentations checkForHits(String response, URI doi, PublicationRepresentations pubRep) {
+        var searchRespnse = toResponse(response);
+        if (searchRespnse.totalHits() > 0) {
```

Comment:
I would extract to method like hasHits() or containsHits(), but it is up to you :)